src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -647,13 +647,15 @@ public synchronized void close(String name) {
   static EpInfo buildEpInfoFromStats(
       long rowCount, Map<String, RowBufferStats> colStats, boolean setDefaultValues) {
     EpInfo epInfo = new EpInfo(rowCount, new HashMap<>());
+    boolean enableDistinctValues = false;
     for (Map.Entry<String, RowBufferStats> colStat : colStats.entrySet()) {
       RowBufferStats stat = colStat.getValue();
+      enableDistinctValues = stat.isEnableDistinctValue();
```

Comment:
this looks odd to me. We are inferring an outer-class characteristic from an inner-object's state.
Lets take in a boolean alongside setDefaultValues

Reply:
infact you can read this.clientBufferParameters.enableDistinctValues and don't even need an argument here.

Reply:
Moved ndv flag to parameter. Cannot use `this.clientBufferParameters` as this is a static method called by chained of static methods.